### PR TITLE
Memoize

### PR DESCRIPTION
**Pull request by VrajPatelK**

### CODE REVIEW

File: Memoize/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+function memoize(fn) {
+
+    // create cache
+    var cache = new Map();
+
+    return function(...args) {
+
+        // if already in cache then return it.
+        if(cache.has(args.toString())){
+            return cache.get(args.toString());
```
Comment (GauravChopada) [inline Memoize/index.js:14]: args.toString() is repeating multiple times, it should be declared once and that variable should be used instead for decreasing toString() function call.

File: Memoize/index.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/**
+ * @param {Function} fn
+ * @return {Function}
+ */
+function memoize(fn) {
+
+    // create cache
+    var cache = new Map();
```
Comment (GauravChopada) [inline Memoize/index.js:8]: If variable is not going to change. Declare it constant

File: Memoize/index.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    return function(...args) {
+
+        // if already in cache then return it.
+        if(cache.has(args.toString())){
+            return cache.get(args.toString());
+        }
+
+        // if it is not then calculate it ad then retunr it. (this time counter will be getting increment)
+        var ans = fn(...args);
```
Comment (GauravChopada) [inline Memoize/index.js:18]: Declare const

**Review by GauravChopada — CHANGES_REQUESTED** [review]

File: Memoize/index.js (hunk 2 above)
Comment (VrajPatelK) [reply Memoize/index.js]: **done! , please verify once.**

File: Memoize/index.js (hunk 3 above)
Comment (VrajPatelK) [reply Memoize/index.js]: **done! , please verify once.**

**Review by GauravChopada — APPROVED** [review]

**Pull request merged**